class-tgm-plugin-activation.php

```diff
@@ -1128,6 +1108,75 @@ public function _get_plugin_data_from_name( $name, $data = 'slug' ) {
 		}
 
 		/**
+		 * Retrieve the download url for a package.
+		 *
+		 * @since 2.5.0
+		 *
+		 * @param string $slug
+		 *
+		 * @return string Plugin download URL or path to local file
+		 */
+		public function get_download_url( $slug ) {
+			$dl = '';
+
+			if ( ! empty( $this->plugins[ $slug ]['source'] ) ) {
+
+				$source = $this->plugins[ $slug ]['source'];
+
+				// Is this a WP repo plugin ?
+				if ( 'repo' === $source || preg_match( self::WP_REPO_REGEX, $source ) ) {
```

Comment:
Is this is now the only usage of the class constants? If so, may be cleaner to just have them as variables in this function - no point in making stuff more public than needs to be.